asyncpg/protocol/protocol.pyx

```diff
@@ -725,6 +725,21 @@ cdef class BaseProtocol(CoreProtocol):
             self.last_query = None
             self.return_extra = False
 
+    cdef _on_notice(self, parsed):
+        # Check it here to avoid unnecessary `_field_map` iteration
+        cb = self.connection._notice_cb_set
+        if cb:
+            fields = apg_exc_base.PostgresMessage._field_map
+            message = {}
+            for k, v in parsed.items():
+                try:
+                    message[fields[k]] = v
+                except KeyError:
+                    # Ignore new unknown fields
+                    pass
+
+            self.connection._notice(message)
```

Comment:
I thought that way, but why don't pass just a dict?
It is easier for logging (people don't need to iterate over an instance of a class), and callbacks don't get anything except a message consists of fields and values.

`PostgresMessage` is designed for exceptions, where it is reasonable to have all known fields as attributes, notices are lightweight, used at specified places (callbacks) and don't need so complex creation process (and reverse process -- iteration via attributes to create a dict to be logged).